devices/HapticGlove/src/HapticGlove.cpp

```diff
@@ -0,0 +1,621 @@
+﻿/**
+ * @file HapticGlove.cpp
+ * @authors  Kourosh Darvish <kourosh.darvish@iit.it>
+ * @copyright 2021 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2021
+ */
+
+
+#include <HapticGlove.h>
+#include <SenseGloveHelper.hpp>
+
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Bottle.h>
+#include <yarp/os/Time.h>
+
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/RpcServer.h>
+
+#include <mutex>
+#include <string>
+#include <stdio.h>
+#include <vector>
+#include <assert.h>
+#include <map>
+
+
+const std::string DeviceName = "HapticGlove";
+const std::string LogPrefix = DeviceName + wearable::Separator;
+double period = 0.01;
+
+using namespace wearable;
+using namespace wearable::devices;
+
+struct SenseGloveIMUData
+{
+    // sensors
+    std::string humanHandLinkName;
+    std::vector<double> humanHandLinkOrientation; // [w, x, y , z]
+
+    std::vector<std::string> humanJointNames;
+    std::vector<double> humanJointValues;
+    std::map<std::string, std::size_t> humanJointSensorNameIdMap;
+
+    // actuators
+    std::vector<double> fingersForceFeedback;
+    std::vector<double> fingersVibroTactileFeedback;
+    std::vector<double> fingersHapticFeedback; // [fingersForceFeedback, fingersVibroTactileFeedback]
+
+    std::vector<std::string> humanFingerNames;
+    std::map<std::string, std::size_t> humanHapticActuatorNameIdMap; // First part is Force Feedback, second part is Vibrotactile feedback
+
+};
+
+class HapticGlove::SenseGloveImpl
+{
+public:
+    mutable std::mutex mutex;
+    SenseGloveIMUData gloveData;
+
+    WearableName wearableName;
+    std::string portsPrefix;
+
+    TimeStamp timeStamp;
+
+    const size_t nFingers = 5;
+
+    // Number of sensors
+    const int nLinkSensors = 1; // 1 palm imu
+
+    // Numbe of actuators
+    const int nActuators = 10; // humanFingerNames.size()*2
+
+    std::unique_ptr<senseGlove::SenseGloveHelper> m_glove; /**< Pointer to the glove object. */
```

Comment:
Can you please uniform the names of the private attributes following the style:

1. If an attribute is declared in the public header of the device, use the `m_` prefix.
2. If an attribute is declared in the implementation class, don't use `m_` since it's redundant. The main class will use `pImpl->privAttr` to access it. Use `this->privAttr` in the Impl methods definition to make it clear that is a class attribute.

Reply:
`this->privAttr` but this is not obligatory, right? I can use the attribute without the `this` pointer , e.g., `privAttr`?

Reply:
Nope, it's just for readability. Using `pImpl->` or `this->` makes clear that it's a class attribute. If I don't see any, I always look for a local variable without finding it. It's not a must, it just improves readability.